src/main/java/org/cryptomator/ui/keyloading/hub/ReceivedKey.java

```diff
@@ -0,0 +1,45 @@
+package org.cryptomator.ui.keyloading.hub;
+
+import com.nimbusds.jose.JWEObject;
+import org.cryptomator.cryptolib.api.Masterkey;
+
+import java.security.interfaces.ECPrivateKey;
+
+@FunctionalInterface
+interface ReceivedKey {
+
+	/**
+	 * Decrypts the vault key.
+	 *
+	 * @param deviceKey This device's private key.
+	 * @return The decrypted vault key
+	 */
+	Masterkey decryptMasterkey(ECPrivateKey deviceKey);
+
+	/**
+	 * Creates an unlock response object from the received legacy "access token" JWE.
```

Comment:
Are you sure this method targets "legacy access tokens"t? I lost a bit the overview 😓

Reply:
copy/paste error in docs

Reply:
fixed in 86f3cb7